server/src/main/java/org/opensearch/search/aggregations/metrics/StarTreeCollector.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.aggregations.metrics;
+
+import org.apache.lucene.index.LeafReaderContext;
+import org.opensearch.index.codec.composite.CompositeIndexFieldInfo;
+import org.opensearch.search.aggregations.bucket.terms.LongKeyedBucketOrds;
+
+import java.io.IOException;
+
+public interface StarTreeCollector {
```

Comment:
Maybe call this `StarTreeAggregator`, since it's expected that `Aggregator` subclasses would implement this interface?